core/src/com/unciv/logic/map/tile/Tile.kt

```diff
@@ -931,6 +934,8 @@ open class Tile : IsPartOfGameInfoSerialization {
         }
 
         owningCity?.reassignPopulationDeferred()
+        if (resource != null && owningCity != null)
+            owningCity!!.civ.cache.updateCivResources()
```

Comment:
Same story here. If the resource is within a city, you should just update resources in case the improvement consumes/provides resources